imod/mf6/wel.py

```diff
@@ -237,6 +243,9 @@ def clip_box(
 
         ds = new.dataset
 
+        z_max = None if layer_max is None else top.isel(layer=layer_max)
```

Comment:
Before this changeset, we expected ``z_max`` and ``z_min`` as a scalar. 
This makes it easy to make simple selections in the table of wells.
This was not feature complete, more a pragmatic workaround so that I not had to deal with locating wells in the grid.

However, top and bottom most of the time are grids. ``z_min``/``z_max`` therefore cannot be expected to be scalar anymore, which means well points have to be located in the grid to do the selecting (at least if ``top`` and ``bottom`` are provided as grids). 
I think we can use ``imod.prepare.wells.locate_wells`` for this. With this we can discard wells easily which are outside top and bottom.

Furthermore: ``top`` can also be provided as scalar by the user, I'm not sure if that perpetuates to the ``clip_box`` method here, or that ``top`` has a got a layer dim assigned to it in the mean time.